apis/config/v1alpha1/projectconfig_types.go

```diff
@@ -41,6 +57,10 @@ type OpenShiftFeatureGates struct {
 	// More details: https://docs.openshift.com/container-platform/latest/networking/understanding-networking.html
 	GatewayRoute bool `json:"gatewayRoute,omitempty"`
 
+	// OpenShiftRoute enables creating OpenShift Route objects instead of Ingress objects.
+	// More details: https://docs.openshift.com/container-platform/latest/networking/understanding-networking.html
+	OpenShiftRoute bool `json:"openshiftRoute,omitempty"`
```

Comment:
Does it only belong to JaegerQuery or to all components that can benefit from ingress/routes?

If its for all components, it might overlaps with:
```golang
	// GatewayRoute enables creating an OpenShift Route for the TempoStack
	// gateway to expose the service to public internet access.
	// More details: https://docs.openshift.com/container-platform/latest/networking/understanding-networking.html
	GatewayRoute bool `json:"gatewayRoute,omitempty"`
```

Reply:
It's similar, yes.
Currently:
* `OpenShiftRoute` allows using Routes instead of Ingress, when a Route/Ingress is enabled in the CR.
* `GatewayRoute` *always* creates a Route for the gateway component for every tempo instance if this feature gate is enabled (with no option to enable/disable the Route in the CR).

Should there be an option in the spec to disable the gateway route per tempo instance? And, should there be an option to use Ingress instead of a Route for the gateway?

Reply:
hm.. I think we should bring the two together. In a way that we can configure the object type and the default behavior per component (UI/Gateway).

Reply:
I've created https://github.com/os-observability/tempo-operator/issues/265 to track this.